src/containers/ToDoList/ToDoList.jsx

```diff
@@ -173,16 +202,24 @@ export const ToDoList = () => {
               </button>
             </p>
           </StyledWrapperButtonSortDates>
-          {sortedAndFilteredTasks.map((task) => (
-            <Task
-              id={task.id}
-              text={task.text}
-              checked={task.checked}
-              onCheck={handleTaskMark}
-              onDelete={handleDeleteTask}
-              dateAdded={task.dateAdded}
-            />
-          ))}
+          <Reorder.Group
+            values={tasks}
+            onReorder={(newTasks) => setTasks(newTasks)}
+            style={{ listStyleType: "none" }}
+          >
+            {sortedAndFilteredTasks.map((task) => (
+              <Reorder.Item value={task} key={task.id}>
+                <Task
+                  id={task.id}
+                  text={task.text}
+                  checked={task.checked}
+                  onCheck={handleTaskMark}
+                  onDelete={handleDeleteTask}
+                  dateAdded={task.dateAdded}
+                />
+              </Reorder.Item>
+            ))}
+          </Reorder.Group>
```

Comment:
That all is responsible for ensuring the possibility of re-ordering the elements -- that right?

Reply:
Yes, both: the reordering and the slide effect when a task is completed/marked